Sparkle/SUTouchBarProvider.m

```diff
@@ -0,0 +1,94 @@
+//
+//  SUTouchBarProvider.m
+//  Sparkle
+//
+//  Created by Yuxin Wang on 05/01/2017.
+//  Copyright © 2017 Sparkle Project. All rights reserved.
+//
+
+#import "SUTouchBarProvider.h"
+
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 101202
+
+@interface SUTouchBarProvider() <NSTouchBarDelegate>
+
+@property (copy) NSString* indentifier;
+@property NSMutableArray<NSTouchBarItem*> *touchBarItems;
+
+@end
+
+@implementation SUTouchBarProvider
+
+@synthesize indentifier;
+@synthesize touchBarItems;
+@synthesize touchBar;
+
+-(instancetype)initWithIdentifier:(NSString *)anIdentifier
+{
+    if (!(self = [super init]))
+        return self;
+    
+    indentifier = [NSString stringWithFormat:@"%@.%@", [NSBundle mainBundle].bundleIdentifier, anIdentifier];
+    touchBarItems = [NSMutableArray arrayWithCapacity:6];
+    
+    touchBar = [[NSTouchBar alloc] init];
+    touchBar.defaultItemIdentifiers = @[indentifier,];
+    touchBar.principalItemIdentifier = indentifier;
+    touchBar.delegate = self;
+    
+    return self;
+}
+
+- (NSTouchBarItem *)touchBar:(NSTouchBar * __unused)touchBar makeItemForIdentifier:(NSTouchBarItemIdentifier)identifier
+{
+    if ([identifier isEqualToString:self.indentifier])
+        return [NSGroupTouchBarItem groupItemWithIdentifier:self.indentifier items:self.touchBarItems];
+    return nil;
+}
+
+-(NSButton *)addButtonWithButton:(NSButton *)button
+{
+    NSString *itemId = [self.indentifier stringByAppendingFormat:@".button-%lu", self.touchBarItems.count];
+    NSCustomTouchBarItem *item = [[NSCustomTouchBarItem alloc] initWithIdentifier:itemId];
+
+    NSButton *buttonCopy = [NSButton buttonWithTitle:button.title target:button.target action:button.action];
+    buttonCopy.keyEquivalent = button.keyEquivalent;
```

Comment:
I prefer to keep it, because the `keyEquivalent` set on returned `NSButton` is a fix for modal window.
I'd like to change the API to `-addButtonWithButton:isDefault:`, to avoid the `keyEquivalent` issue.